solidity/dashboard/src/services/liquidity-rewards.js

```diff
@@ -159,6 +167,46 @@ class UniswapLPRewards extends LiquidityRewards {
 
     return this._calculateAPY(r, WEEKS_IN_YEAR)
   }
+
+  /**
+   * Calculates lp token balance for the given pair
+   * The calculations were done based on
+   * https://uniswap.org/docs/v2/advanced-topics/understanding-returns/#why-is-my-liquidity-worth-less-than-i-put-in
+   *
+   * @param {string} lpBalance Balance of liquidity token for a given uniswap pair deposited in
+   * the LPRewards` contract.
+   * @return {Promise<{token0: string, token1: string}>}
+   */
+  calculateLPTokenBalance = async (lpBalance) => {
+    const pairData = await getPairData(this.wrappedTokenAddress.toLowerCase())
+
+    const shareOfUniswapPool = percentageOf(
+      lpBalance.toString(),
+      fromTokenUnit(pairData.totalSupply).toString()
+    )
+
+    const constantProduct = new BigNumber(pairData.reserve0).multipliedBy(
+      pairData.reserve1
+    )
+
+    const token0LiquidityPool = constantProduct
+      .div(pairData.token1Price)
+      .squareRoot()
+    const token1LiquidityPool = constantProduct
+      .multipliedBy(pairData.token1Price)
```

Comment:
Shouldn't be `pairData.token0Price`?

Reply:
I think it should not. Based on [Uniswap docs](https://uniswap.org/docs/v2/advanced-topics/understanding-returns/#why-is-my-liquidity-worth-less-than-i-put-in) we are using the price of the second token (token1) and we use it to calculate `token0LiquidityPool` and `token1LiquidityPool`.

Reply:
You're right! 🚀